### ISSUE TITLE
channel Receiver docs say it returns None when all senders have been dropped, but this is impossible

### ISSUE THREAD

**Issue by kellpossible** [async-rs/async-std#850]
https://docs.rs/async-std/1.6.2/async_std/sync/struct.Receiver.html#impl

Docs for `recv()`:

> Receives a message from the channel.
>
> If the channel is empty and still has senders, this method will wait until a message is sent into it. Once all senders have been dropped it will return `None`.

How can it return `None` when the type signature is `pub async fn recv<'_>(&'_ self) -> Result<T, RecvError>`? It appears to be throwing an error instead, and an opaque error type too which makes it impossible to respond correctly to the specific situation where the senders were dropped vs some other kind of error condition.

This code triggers a panic on the last line `panicked at 'The channel is empty.'`:

```rust
let (tx, rx) = async_std::sync::channel(10);
tx.send("hello").await;
drop(tx);
let msg = rx.recv().await.unwrap();
assert_eq!("hello", msg);
let result= rx.recv().await;
panic!(result.unwrap_err().to_string());
```

**Comment by kellpossible** [comment]
The [async-std book tutorial](https://book.async.rs/tutorial/sending_messages.html) recommends using [futures::channel::mpsc](https://docs.rs/futures/0.3.5/futures/channel/mpsc/index.html). What is the difference between this, and the `async-std` channel, and which should be preferred?

**Comment by kellpossible** [comment]
It looks like this issue may be solved by https://github.com/async-rs/async-std/commit/cd7fb9dec248fc6a79d3af96cc86297363b3a5be

**Comment by yoshuawuyts** [comment]
Closed by #827 which adds a new channel impl
